pac4j-core/src/main/java/org/pac4j/core/credentials/authenticator/LocalCachingAuthenticator.java

```diff
@@ -56,8 +59,30 @@ public void validate(final T credentials, final WebContext context) throws HttpA
             });
             credentials.setUserProfile(profile);
             logger.debug("Found cached credential. Using cached profile {}...", profile);
-        } catch (Exception e) {
-            throw new CredentialsException(e);
+        } catch (UncheckedExecutionException e) {
+            if (e.getCause() instanceof RuntimeException) {
+                throw (RuntimeException) e.getCause();
+            } else if (e.getCause() != null) {
```

Comment:
I take your point completely. There is inherent danger in "forgetting" to put in the proper return statements. No doubt. Else clauses help. Unit tests do too! My comment isn't so much about which is better; it's more about sticking to one single consistent style across the board. In general, it's lots easier to say "This is our style, deal with it" rather than launching into some debate about the metrics of compiler efficiency, etc :) I simply advocate for a style that helps me do/write less. :D

@leleuj that might be something we could put into our styling rules here; Pick one and apply across the board.